### DISCUSSION TITLE
NiceGUI forms sharing data with all users

### DISCUSSION THREAD

**Discussion by kgfelix** [zauberzeug/nicegui#2049] [General]
Hello Team.
I created an application with NiceGUI. However, when testing in staging I realized that the form data is shared with all users who access the application.

For example, I have an input field "Supplier Id":
` survey_id = ui.input(
            label="Survey ID",
            validation={"Input too long": lambda value: len(value) < 20},
        )`
 and if another user adds the value "123" to the form in another browser, it is possible to see the same information.

I already checked the "app.storage" and tested with and without it, but I have the same behavior.
Has anyone experienced this and know how to resolve it?

Thanks,

K. Felix

**Comment by natankeddem** [comment]
https://github.com/zauberzeug/nicegui/wiki/FAQs#why-are-all-browser-tabswindows-showing-the-same-content
